utils/v2tofhirr4/utilities.bal

```diff
@@ -110,45 +113,62 @@ public isolated function pidToPatientName(hl7v2commons:Pid5 pid5, hl7v2commons:P
         }
     }
 
-    return humanNames;
+    return (humanNames.length() > 0) ? humanNames : ();
 }
 
-public isolated function pidToAddress(string pid12, hl7v2commons:Pid11 pid11) returns r4:Address[] {
+public isolated function pidToAddress(string pid12, hl7v2commons:Pid11 pid11) returns r4:Address[]? {
     r4:Address[] address = [];
     if pid12 != "" {
         address.push({district: pid12});
     }
 
     if pid11 is hl7v23:XAD[] {
         foreach hl7v23:XAD item in pid11 {
-            address.push(xadToAddress(item));
+            r4:Address? xadToAddressResult = xadToAddress(item);
+            if xadToAddressResult is r4:Address {
+                address.push(xadToAddressResult);
+            }
         }
     } else if pid11 is hl7v24:XAD[] {
         foreach hl7v24:XAD item in pid11 {
-            address.push(xadToAddress(item));
+            r4:Address? xadToAddressResult = xadToAddress(item);
+            if xadToAddressResult is r4:Address {
+                address.push(xadToAddressResult);
+            }
         }
     } else if pid11 is hl7v25:XAD[] {
         foreach hl7v25:XAD item in pid11 {
-            address.push(xadToAddress(item));
+            r4:Address? xadToAddressResult = xadToAddress(item);
+            if xadToAddressResult is r4:Address {
+                address.push(xadToAddressResult);
+            }
         }
     } else if pid11 is hl7v26:XAD[] {
         foreach hl7v26:XAD item in pid11 {
-            address.push(xadToAddress(item));
+            r4:Address? xadToAddressResult = xadToAddress(item);
+            if xadToAddressResult is r4:Address {
+                address.push(xadToAddressResult);
+            }
         }
     } else if pid11 is hl7v27:XAD[] {
         foreach hl7v27:XAD item in pid11 {
-            address.push(xadToAddress(item));
+            r4:Address? xadToAddressResult = xadToAddress(item);
+            if xadToAddressResult is r4:Address {
+                address.push(xadToAddressResult);
+            }
         }
     } else if pid11 is hl7v28:XAD[] {
         foreach hl7v28:XAD item in pid11 {
-            address.push(xadToAddress(item));
+            r4:Address? xadToAddressResult = xadToAddress(item);
+            if xadToAddressResult is r4:Address {
+                address.push(xadToAddressResult);
+            }
         }
     }
-
-    return address;
+    return (address != []) ? address : ();
```

Comment:
Should we check the length of the array instead of doing an equality?

Reply:
+1